src/OrleansRuntime/Startup/StartupBuilder.cs

```diff
@@ -12,89 +9,13 @@ namespace Orleans.Runtime.Startup
     /// </summary>
     internal class StartupBuilder
     {
-        internal static IServiceProvider ConfigureStartup(string startupTypeName, out bool usingCustomServiceProvider)
-        {
-            usingCustomServiceProvider = false;
-            IServiceCollection serviceCollection = new ServiceCollection();
-            ConfigureServicesBuilder servicesMethod = null;
-            Type startupType = null;
-
-            if (!String.IsNullOrWhiteSpace(startupTypeName))
-            {
-                startupType = Type.GetType(startupTypeName);
-                if (startupType == null)
-                {
-                    throw new InvalidOperationException($"Can not locate the type specified in the configuration file: '{startupTypeName}'.");
-                }
-
-                servicesMethod = FindConfigureServicesDelegate(startupType);
-                if (servicesMethod != null && !servicesMethod.MethodInfo.IsStatic)
-                {
-                    usingCustomServiceProvider = true;
-                }
-            }
-
-            RegisterSystemTypes(serviceCollection);
-
-            if (usingCustomServiceProvider)
-            {
-                var instance = Activator.CreateInstance(startupType);
-                return servicesMethod.Build(instance, serviceCollection);
-            }
-
```

Comment:
The addition of the `BuildServiceProvider` method on `SiloHostBuilder` is where you can use the `IServiceCollection` to use a 3rd-party DI container instead of the default one.